sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1177,11 +1201,13 @@ def task_worker(self):
                     # Skip if XcvrApi is not supported
                     api = sfp.get_xcvr_api()
                     if api is None:
+                        self.log_error("{}: skipping CMIS state machine since no xcvr api!!!".format(lport))
                         self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
                         continue
 
                     # Skip if it's not a paged memory device
                     if api.is_flat_memory():
+                        self.log_notice("{}: skipping CMIS state machine for flat memory xcvr".format(lport))
```

Comment:
You may be periodically logging this, then it is not right.

Reply:
the state machine moves to CMIS_STATE_READY after this, and doesnot come again here unless the state is changed.